src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetDomainName.cs

```diff
@@ -31,7 +31,13 @@ internal static unsafe string GetDomainName()
                 throw new InvalidOperationException($"{nameof(GetDomainName)}: {err}");
             }
 
-            return Marshal.PtrToStringUTF8((IntPtr)name)!;
+            string domainName = Marshal.PtrToStringUTF8((IntPtr)name)!;
+            if (domainName == "(none)")
```

Comment:
In what situation / on what platforms does getdomainname return "(none)"?

Reply:
that feel like stringification of `null`

Reply:
I check I get the string `(none)` directly from `getdomainname` on my Ubuntu VM. (no NIS configured e.g. default) Return value is 0 e.g. success.
on macOS I get empty string. 

So this change tries to interpret result returned by the underlying OS function. I did not test if that changes if you change OS locale.

Reply:
It does not seems to depend on locale
```
furt@kamna:~/projects/newtorkInfo$ ~/dotnet-latest/dotnet run
Obnovení dokončeno (0,5s)
Používáte verzi Preview rozhraní .NET. Viz: https://aka.ms/dotnet-support-policy
  newtorkInfo úspěšné (0,2s) → bin/Debug/net9.0/newtorkInfo.dll

Sestavení úspěšné za 1,0s
Name kamna domain (none)
```

there is some explanation here:
https://access.redhat.com/solutions/352963

and  Kernel github

https://github.com/torvalds/linux/blob/e32cde8d2bd7d251a8f9b434143977ddf13dcec6/include/linux/uts.h#L17